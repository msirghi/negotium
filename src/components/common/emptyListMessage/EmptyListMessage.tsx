import { FC } from 'react';
import { useEmptyListMessageStyles } from './styles';

type Props = {
  imageSrc: string;
  message: string;
};

export const EmptyListMessage: FC<Props> = ({ imageSrc, message }) => {
  const classes = useEmptyListMessageStyles();

  return (
    <>
      <img src={imageSrc} className={classes.image} alt={imageSrc} />
      <div className={classes.message}>{message}</div>
    </>
  );
};
