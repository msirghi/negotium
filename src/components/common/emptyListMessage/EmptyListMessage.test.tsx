import renderer from 'react-test-renderer';
import { EmptyListMessage } from './EmptyListMessage';
import { render } from '@testing-library/react';

describe('EmptyListMessage', () => {
  const defaultProps = {
    imageSrc: 'src',
    message: 'Message',
  };

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should match the snapshot', () => {
    const tree = renderer.create(<EmptyListMessage {...defaultProps} />);
    expect(tree).toMatchSnapshot();
  });

  it('should render provided text', () => {
    const { getByText } = render(<EmptyListMessage {...defaultProps} />);
    expect(getByText(defaultProps.message)).toBeInTheDocument();
  });
});
