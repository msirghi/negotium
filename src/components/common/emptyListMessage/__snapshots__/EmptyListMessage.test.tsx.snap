// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmptyListMessage should match the snapshot 1`] = `
Array [
  <img
    alt="src"
    className="makeStyles-image-1"
    src="src"
  />,
  <div
    className="makeStyles-message-2"
  >
    Message
  </div>,
]
`;
